packages/fx-core/src/component/generator/copilotExtension/generator.ts

```diff
@@ -52,12 +52,11 @@ const enum telemetryProperties {
 export class CopilotExtensionGenerator extends DefaultTemplateGenerator {
   componentName = "copilot-extension-from-scratch-generator";
   public activate(context: Context, inputs: Inputs): boolean {
-    return (
-      (inputs[QuestionNames.Capabilities] === CapabilityOptions.declarativeAgent().id &&
-        inputs[QuestionNames.ApiPluginType] !== ApiPluginStartOptions.apiSpec().id) ||
-      (inputs[QuestionNames.Capabilities] === CapabilityOptions.apiPlugin().id &&
-        inputs[QuestionNames.ApiPluginType] === ApiPluginStartOptions.newApi().id)
-    );
+    return [
+      TemplateNames.ApiPluginFromScratch,
+      TemplateNames.ApiPluginFromScratchBearer,
+      TemplateNames.ApiPluginFromScratchOAuth,
+    ].includes(inputs[QuestionNames.TemplateName]);
```

Comment:
Missing TemplateNames.BasicGpt. If basic gpt template + no plugin, post() should not run (the default generator should be enough). However, If basic gpt template + existing plugin, post() need to be run.